tasty-plutus/src/Test/Tasty/Plutus/Internal/DumpScript.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+
+module Test.Tasty.Plutus.Internal.DumpScript (
+  DumpScript (DumpScript),
+) where
+
+import Data.Coerce (Coercible, coerce)
+import Data.Kind (Type)
+import Data.Proxy (Proxy (Proxy))
+import Data.Tagged (Tagged (Tagged))
+import Data.Text.IO qualified as Text
+import Plutus.V1.Ledger.Scripts (Script (unScript))
+import PlutusCore.Pretty qualified as PLC
+import PlutusTx (CompiledCode, getPir, getPlc)
+import System.FilePath ((</>))
+import Test.Tasty.Options (
+  IsOption,
+  OptionDescription (Option),
+  OptionSet,
+  lookupOption,
+ )
+import Test.Tasty.Plutus.Internal.Options (
+  DumpPath (DumpPath, NoDumpPath),
+  DumpPlutusCore (DumpPlutusCore),
+  DumpPlutusIR (DumpPlutusIR),
+  DumpPlutusTypedCore (DumpPlutusTypedCore),
+ )
+import Test.Tasty.Providers (IsTest (run, testOptions), testPassed)
+
+runIf ::
+  forall (v :: Type).
+  (IsOption v, Coercible v DumpPath) =>
+  OptionSet ->
+  (FilePath -> IO ()) ->
+  IO ()
+runIf opts action = case coerce (lookupOption @v opts) of
+  DumpPath path -> action path
+  NoDumpPath -> pure ()
+
+data DumpScript = forall a. DumpScript String Script (CompiledCode a)
```

Comment:
I'd prefer this written as a GADT:

```haskell
data DumpScript where
      DumpScript :: String -> Script -> CompiledCode a -> DumpScript
```

Reply:
Out of curiosity: Why?

Reply:
Mostly because I find it easier. I'm not particularly opinionated on this though.